docs/robot/virtual-robot/robot-class.md

```diff
@@ -124,62 +126,397 @@ public class ObstacleAvoidRobot extends VirtualRobot {
 
 ### ProximitySensor(Robot robot, RobotMqttClient m) 
 
-#### getProximity() -> double
+    ProximitySensor proximitySensor = new ProximitySensor(this, robotMqttClient);
+
+#### getProximity()
+
+<span style="{{%sub_color%}}">Description</span>
+
+Returns ProximityReadingType value indicating .......
+    
+<span style="{{%sub_color%}}">Syntax</span>
+
+	proximitySensor.getProximity();
+
+<span style="{{%sub_color%}}">Parameter</span>
+
+None
+
+<span style="{{%sub_color%}}">Returns</span>
+
+swarm.robot.types.ProximityReadingType
 
 ### ColorSensor(Robot robot, RobotMqttClient m) 
 
-#### getColor() -> RGBColorType
+    ColorSensor colorSensor = new ColorSensor(this, robotMqttClient);
+
+#### getColor() 
+
+<span style="{{%sub_color%}}">Description</span>
+
+Returns RGBColor type value indicating the color detected at the front of the robot.
+    
+<span style="{{%sub_color%}}">Syntax</span>
+ 
+	colorSensor.getColor() 
+
+<span style="{{%sub_color%}}">Parameter</span>
+
+None
 
+<span style="{{%sub_color%}}">Returns</span>
 
+swarm.robot.types.RGBColorType 
 
 ## Communication
 
 ### DirectedCommunication(int robotId, RobotMqttClient m)
 
+    DirectedCommunication directedComm = new DirectedCommunication(id, robotMqttClient);
 
+#### sendMessage(String msg, int distance)
 
 
+<span style="{{%sub_color%}}">Description</span>
+
+Sends message through MQTT.
```

Comment:
Sends a broadcast type message to the robots in the front of the robot within [-15,15] angle